runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -396,6 +384,80 @@ public void onChatMessage(ChatMessage event)
 		{
 			createGameTimer(PRAYER_ENHANCE);
 		}
+
+		if (config.showHomeTeleport() && event.getMessage().startsWith("You need to wait another ") && event.getMessage().endsWith(" to cast this spell."))
+		{
+			long time;
+			if (event.getMessage().equals("You need to wait another minute to cast this spell."))
+			{
+				time = 1;
+			}
+			else
+			{
+				time = Long.parseLong(event.getMessage().split(" ")[5]);
+			}
+			createGameTimerWithVariableTime(HOME_TELEPORT, time, ChronoUnit.MINUTES);
+		}
+
+		if (config.showMinigameTeleport() && event.getMessage().startsWith("You must wait another ") && event.getMessage().endsWith(" before you can use the minigame teleports."))
+		{
+			long time;
+			if (event.getMessage().equals("You must wait another minute before you can use the minigame teleports."))
+			{
+				time = 1;
+			}
+			else
+			{
+				time = Long.parseLong(event.getMessage().split(" ")[4]);
+			}
+			createGameTimerWithVariableTime(MINIGAME_TELEPORT, time, ChronoUnit.MINUTES);
+		}
+	}
+
+	@Subscribe
+	public void onTick(GameTick event)
+	{
+		if (checkTeleportDestination)
+		{
+			//Make sure the check isn't running forever if we manage to cancel the teleport
+			if (checksLeft == 0)
+			{
+				checkTeleportDestination = false;
+				return;
+			}
+			checksLeft -= 1;
+
+			boolean moved = false;
+			WorldPoint currentPosition = client.getLocalPlayer().getWorldLocation();
+			if (currentPosition.getRegionID() != lastWorldLocation.getRegionID())
+			{
+				moved = true;
+			}
+			else
+			{
+				if (homeTeleportRegions.contains(currentPosition.getRegionID()) || minigameTeleportRegions.contains(currentPosition.getRegionID()))
+				{
+					if (currentPosition.distanceTo(lastWorldLocation) > 1)
```

Comment:
I don't think this will work very well since you can start the anim and just run away and it would trigger.

Reply:
Whoops, I see what you mean. 

It was only an issue on the Cow Teleport, as you can't move during the final part of the normal home teleport animation.